### ISSUE TITLE
Use profile defined formats instead of language for formatting. 

### ISSUE THREAD

**Issue by MrEbbinghaus** [alexarch21/history-explorer-card#221]
In Home Assistant, users can set number/date/time formats independent of the language, as well as the display timezone (local time/server time). Those should be respected. 

Currently, only the settings for 12/24-hour time format get somewhat respected. 

The different possible formats can be found here:
https://github.com/home-assistant/frontend/blob/a5e5efa5263de790f4624a581f71d6db35eefa65/src/data/translation.ts#L4-L30

To get the system locale and timezone, you can call `DateTimeFormat` with `undefined`: 

`Intl.DateTimeFormat(undefined).resolvedOptions().locale`

This is related to (but not the same as): https://github.com/alexarch21/history-explorer-card/issues/219

**Comment by alexarch21** [comment]
Thanks for the contribution.

I can see value in adding support for language independent formatting to the card. There are however a number of limiting factors that need to be considered. 

Tick time / date format, for example, is intentionally kept non-configurable with hardcoded date formats, as anything longer can lead to massive rendering artifacts with tick labels overlapping. While the backend graph rendering engine, chartjs, does offer several ways to adjust layout to longer tick labels avoiding overlap, this is incompatible with the way the engine is (ab)used by this card for realtime smooth scrolling and will lead to jumping graphs and changing layout while scrolling. The hardcoded formats offer consistency in scrolling behavior.

styleDateSelector and styleDateTimeTooltip are less problematic in that respect.

Display timezone is non-trivial to change and will require adjustments throughout the card related to how time is managed locally and in conjunction with remote database queries.

And finally, full backwards compatibility with older HA versions that may not support the separate format selection will have to be ensured.

In conclusion, yes, adding support for user defined formats from within HA is a good idea. But it must adhere to the aforementioned limitations and will have to be tested against them. I will test your PR and outline changes as needed.

**Comment by alexarch21** [comment]
Sadly the history explorer card has been declared end of life and won't be developed anymore. So I will have to close this issue / FR without resolution. This repository will be archived and set to read-only shortly.

See this post on the HA forum for more information: 

https://community.home-assistant.io/t/new-interactive-history-explorer-custom-card/369450/978
